src/Nerdbank.MessagePack/Converters/ObjectMapConverter`1.cs

```diff
@@ -38,36 +52,43 @@ internal class ObjectMapConverter<T>(
 	public override void Write(ref MessagePackWriter writer, in T? value, SerializationContext context)
 #pragma warning restore NBMsgPack031
 	{
-		if (value is null)
+		try
```

Comment:
Make similar changes to the ObjectMapWithNonDefaultCtorConverter and ObjectArrayWithNonDefaultCtorConverter.